cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -462,11 +466,27 @@ public void testMapperDatasetAccess() throws Exception {
 
     ApplicationManager appManager = deployApplication(DatasetWithMRApp.class);
     Map<String, String> argsForMR = ImmutableMap.of(DatasetWithMRApp.INPUT_KEY, "table1",
-                                                    DatasetWithMRApp.OUTPUT_KEY, "table2");
+                                                    DatasetWithMRApp.OUTPUT_KEY, "table2",
+                                                    Constants.Metrics.EMIT_MR_TASKL_LEVEL_METRICS, "false");
     MapReduceManager mrManager = appManager.getMapReduceManager(DatasetWithMRApp.MAPREDUCE_PROGRAM).start(argsForMR);
     mrManager.waitForRun(ProgramRunStatus.COMPLETED, 5, TimeUnit.MINUTES);
-    appManager.stopAll();
 
+    appManager.stopAll();
+    List<TagValue> tags = new ArrayList<>();
+    tags.add(new TagValue(Constants.Metrics.Tag.NAMESPACE, NamespaceId.DEFAULT.getNamespace()));
+    tags.add(new TagValue(Constants.Metrics.Tag.APP, DatasetWithMRApp.class.getSimpleName()));
+    tags.add(new TagValue(Constants.Metrics.Tag.MAPREDUCE, DatasetWithMRApp.MAPREDUCE_PROGRAM));
+    tags.add(new TagValue(Constants.Metrics.Tag.DATASET, "table1"));
+    tags.add(new TagValue(Constants.Metrics.Tag.RUN_ID, mrManager.getHistory().get(0).getPid()));
+    Collection<TagValue> tagValues =
+      getMetricsManager().searchTags(new MetricSearchQuery(0, Integer.MAX_VALUE, Integer.MAX_VALUE, tags));
+    // map only task, search should have returned mapper task type
+    Assert.assertEquals(1, tagValues.size());
+    tags.add(tagValues.iterator().next());
+    tagValues =
+      getMetricsManager().searchTags(new MetricSearchQuery(0, Integer.MAX_VALUE, Integer.MAX_VALUE, tags));
```

Comment:
see above. You need to run the same test with task level metrics enabled, to validate that that case also works.